src/quo/components/drawers/permission_context/view.cljs

```diff
@@ -1,13 +1,101 @@
-   children])
+    [quo.theme]
+    [react-native.blur :as blur]
+    [react-native.core :as rn]
+    [react-native.platform :as platform]
+    [react-native.shadow :as shadow]
+    [schema.core :as schema]
+    [utils.i18n :as i18n]))
+
+(defn- single-token-gating
+  [{:keys [blur? token-value token-symbol]}]
+  [rn/view
+   {:style               {:flex-direction :row}
+    :accessibility-label :permission-context-single-token}
+   [text/text {:style {:margin-right 3}}
+    (i18n/label :t/hold-to-post-1)]
+   [token-tag/view
+    {:size         :size-24
+     :option       false
+     :blur?        blur?
+     :token-value  token-value
+     :token-symbol token-symbol}]
+   [text/text {:style {:margin-left 3}}
+    (i18n/label :t/hold-to-post-2)]])
+
+(defn- token-group
+  [blur? idx group]
+  ^{:key idx}
+  [rn/view {:style style/token-group}
+   (when-not (= idx 0)
+     [text/text {:style {:margin-right 3}}
+      (string/lower-case (i18n/label :t/or))])
+   [preview-list/view
+    {:type  :tokens
+     :blur? blur?
+     :size  :size-24} group]])
+
+(defn- multiple-token-gating
+  [{:keys [token-groups blur?]}]
+  (let [visible-token-groups (take 2 token-groups)
+        extra-groups?        (-> token-groups count (> 2))]
+    [rn/view
+     {:style               {:flex-direction :row}
+      :accessibility-label :permission-context-multiple-token}
+     [text/text (i18n/label :t/hold-to-post-1)]
+     [rn/view {:style {:flex-direction :row}}
+      (map-indexed (partial token-group blur?) visible-token-groups)]
+     (when extra-groups?
+       [:<>
+        [text/text {:style {:margin-horizontal 3}}
+         (string/lower-case (i18n/label :t/or))]
+        [number-tag/view
+         {:size   :size-24
+          :number "9999" ;; show the options tag
+          :blur?  blur?
+          :type   :rounded}]])
+     [text/text {:style {:margin-left 3}}
+      (i18n/label :t/hold-to-post-2)]]))
+
+(defn- view-internal
+  [{:keys [on-press blur? type container-style] :as props}]
+  (let [theme (quo.theme/use-theme-value)]
+    [shadow/view
+     {:offset       [0 4]
+      :paint-inside false
+      :start-color  (colors/theme-colors colors/neutral-100-opa-8 colors/neutral-100-opa-60)
+      :distance     25
+      :style        {:align-self :stretch}}
+     [rn/view {:style (merge (style/container blur? theme) container-style)}
+      (when blur?
+        [blur/view
+         {:style         style/blur-container
+          :blur-amount   20
+          :blur-radius   (if platform/ios? 20 10)
+          :overlay-color (colors/theme-colors colors/white-70-blur colors/neutral-95-opa-70-blur)
+          :blur-type     :transparent}])
+      [button/button
+       {:type      :ghost
+        :size      24
+        :on-press  on-press
+        :icon-left (when (= type :action)
+                     (:action-icon props))}
+       (condp = type
+         :action                (:action-label props)
```

Comment:
action-label can be destructured.

Reply:
I did this on purpose, as `:action-label` and `:action-icon` are specific to the `:action` type. Same for the other types, just there I used `select-keys` to extract the prop.